libcontainer/hooks.go

```diff
@@ -0,0 +1,68 @@
+package libcontainer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+func runHook(hook spec.Hook, cid, bundlePath string) error {
+	// Adapted from:
+	// github.com/kata-containers/runtime/cli/hook.go
+	state := spec.State{
+		Pid:    os.Getpid(),
+		Bundle: bundlePath,
+		ID:     cid,
+	}
+
+	stateJSON, err := json.Marshal(state)
+	if err != nil {
+		return err
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := &exec.Cmd{
```

Comment:
it's interesting how starting a container needs to start at least 2 processes during setup.